exonum/src/blockchain/mod.rs

```diff
@@ -349,6 +349,10 @@ impl Blockchain {
                 for precommit in precommits {
                     schema.precommits_mut(&block_hash).push(precommit.clone());
                 }
+
+                // Consensus messages cache is useful only during one height, so it should be
+                // cleared when a new height is achieved.
+                schema.consensus_messages_cache_mut().clear();
```

Comment:
Is the message cache only cleared here? If so, does it mean that a node may process the same message several times if it crashes repeatedly as per [`NodeHandler::initialize()`](https://github.com/skletsun/exonum/blob/49b05f37dced3341a4f709c7602c09f9d3bbb4e9/exonum/src/node/mod.rs#L505)?

Reply:
Right. Why this is a problem?

Reply:
I mean how can it be avoided? On the one  hand, node can become byzantine if it doesn't store its messages. On the other hand, it can keep crashing forever.

Probably we can provide some interface to explicitly clear cached messages during initialization. What do you think?